ui/src/app/pages/admin/admin-user-profile.spec.tsx

```diff
@@ -0,0 +1,492 @@
+      updateCache,
+    });
+
+    registerApiClient(EgressEventsAdminApi, new EgressEventsAdminApiStub());
+    registerApiClient(UserAdminApi, new UserAdminApiStub());
+    registerApiClient(InstitutionApi, new InstitutionApiStub());
+  });
+
+  it('should render', () => {
+    const wrapper = component();
+    expect(wrapper).toBeTruthy();
+  });
+
+  it("should display the user's name, username, and initial credits usage", async () => {
+    const givenName = 'John Q';
+    const familyName = 'Public';
+    const expectedFullName = 'John Q Public';
+
+    const username = 'some-email@yahoo.com';
+
+    const freeTierUsage = 543.21;
+    const freeTierDollarQuota = 678.99;
+    const expectedCreditsText = '$543.21 used of $678.99 limit';
+
+    updateTargetProfile({
+      username,
+      givenName,
+      familyName,
+      freeTierUsage,
+      freeTierDollarQuota,
+    });
+
+    const wrapper = component();
+    expect(wrapper).toBeTruthy();
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(getUneditableFieldText(wrapper, 'name')).toEqual(expectedFullName);
+    expect(getUneditableFieldText(wrapper, 'user-name')).toEqual(username);
+    expect(getUneditableFieldText(wrapper, 'initial-credits-used')).toEqual(
+      expectedCreditsText
+    );
+  });
+
+  test.each([
+    [
+      'RT only',
+      [AccessTierShortNames.Registered],
+      AccessTierDisplayNames.Registered,
+    ],
+    [
+      'CT only',
+      [AccessTierShortNames.Controlled],
+      AccessTierDisplayNames.Controlled,
+    ],
+    [
+      'RT and CT',
+      [AccessTierShortNames.Registered, AccessTierShortNames.Controlled],
+      'Registered Tier, Controlled Tier',
+    ],
+    ['neither', [], 'No data access'],
+  ])(
+    'should display access tiers if the user has membership in %s',
+    async (_, accessTierShortNames, expectedText) => {
+      updateTargetProfile({ accessTierShortNames });
+
+      const wrapper = component();
+      expect(wrapper).toBeTruthy();
+      await waitOneTickAndUpdate(wrapper);
+
+      expect(getUneditableFieldText(wrapper, 'data-access-tiers')).toEqual(
+        expectedText
+      );
+    }
+  );
+
+  it('should allow updating contactEmail within an institution', async () => {
+    updateTargetProfile({ contactEmail: BROAD_ADDR_1 });
+
+    const wrapper = component();
+    expect(wrapper).toBeTruthy();
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(findTextInput(wrapper, 'contactEmail').props().value).toEqual(
+      BROAD_ADDR_1
+    );
+
+    await simulateTextInputChange(
+      findTextInput(wrapper, 'contactEmail'),
+      BROAD_ADDR_2
+    );
+    expect(findTextInput(wrapper, 'contactEmail').props().value).toEqual(
+      BROAD_ADDR_2
+    );
+    expect(wrapper.find('[data-test-id="email-invalid"]').exists()).toBeFalsy();
+
+    const saveButton = wrapper.find('[data-test-id="update-profile"]');
+    expect(saveButton.exists()).toBeTruthy();
+    expect(saveButton.props().disabled).toBeFalsy();
```

Comment:
Is `.exists()` check redundant? If it is missing, doesn't `saveButton.props()` throws exception?

Reply:
if the saveButton is not found, I think this error would be clearer

Reply:
👍🏻